code/game/objects/items/weaponry.dm

```diff
@@ -110,6 +108,7 @@ for further reading, please see: https://github.com/tgstation/tgstation/pull/301
 /obj/item/claymore/highlander/Initialize()
 	. = ..()
 	ADD_TRAIT(src, TRAIT_NODROP, HIGHLANDER_TRAIT)
+	AddElement(/datum/element/update_icon_updates_onmob)
```

Comment:
this won't do anything since you're not using the update_icon proc as you should be to change icon_states.